lib/client/signalflow/signalflow_client.js

```diff
@@ -74,6 +74,52 @@ function SignalflowClient(apiToken, options) {
     };
   }
 
+  function liveTailRequest(opts) {
+    var msgBuffer = [];
+    var callback = null;
+
+    function resolveMessage(msg) {
+      if (msg.type === 'error') {
+        errorMessageCallback(msg);
+      } else {
+        callback(null, msg);
+      }
+    }
+
+    function errorMessageCallback(msg) {
+      callback(msg, null);
+    }
+
+    function msgCallback(msg) {
+      if (!callback) {
+        msgBuffer.push(msg);
+      } else {
+        resolveMessage(msg);
+      }
+    }
+
+    var callbacks = {
+      onMessage: msgCallback,
+      onError: errorMessageCallback
```

Comment:
The request manager doesn't actually rely on your `onError` callback, so you don't need to pass it in. See my other comment about the method definition of `registerLiveTail()` anyway.

Reply:
I want to add this support in the future once the backend implements errors, so I left it here

Reply:
You misunderstood my point: you're taking care of that already at this layer in `resolveMessage()`.

Reply:
I think I got what you are saying. I cleanup up the callbacks, please take a look at my latest commit